docs/benchmarks.md

```diff
@@ -0,0 +1,125 @@
+scale factor, where the scale factor is an integer representing approximately how many gigabytes 
+of data will be generated, with scale factor 1 meaning ~1 GB, and scale factor 1000 meaning ~1 TB, 
+for example.
+
+Further information on data generation can be found using the following links:
+
+- [TPC-DS Data Generator](https://github.com/databricks/tpcds-kit)
+- [TPC-H Data Generator](https://github.com/electrum/tpch-dbgen)
+- [TPCx-BB Data Generator](http://www.tpc.org/tpc_documents_current_versions/current_specifications5.asp)
+
+The remainder of this document is based on the TPC-DS benchmark but the steps are very similar for 
+the other benchmarks.
+
+## Spark Shell
+
+The integration test jar needs to be added to the `--jars` configuration option when launching the
+Spark shell. This jar can be found in the `integration_tests/target` directory after running
+`mvn package`, with a filename matching `rapids-4-spark-integration-tests_2.12-*-SNAPSHOT.jar`.
+
+To run benchmarks on the GPU, the RAPIDS Accelerator for Apache Spark must also be installed,
+following the instructions provided in the [Getting Started](get-started/getting-started.md) guide.
+
+## Converting to Parquet
+
+Although it is possible to run benchmarks directly against the CSV data generated by the TPC data 
+generators, it is common to convert the data to Parquet format and run benchmarks against the 
+Parquet files instead.
+
+The `integration_test` module contains code for converting the CSV data sets to Parquet.
+
+The following commands can be entered into spark-shell to perform the conversion.
+
+```
+import com.nvidia.spark.rapids.tests.tpcds._
+TpcdsLikeSpark.csvToParquet(spark, "/path/to/input", "/path/to/output")
+```
+
+Note that the code for converting CSV to Parquet does not explicitly specify the number of 
+partitions to write, so the size of the resulting parquet files will vary depending on the value 
+for `spark.default.parallelism`, which by default is based on the number of available executor 
+cores. This value can be set explicitly to better control the size of the output files.
+
+It should also be noted that no decimal types will be output. The conversion code uses explicit 
+schemas to ensure that decimal types are converted to floating-point types instead.
+
+## Running Benchmarks
+
+The benchmarks can be executed in two modes currently:
+
+- Execute the query and collect the results to the driver
+- Execute the query and write the results to disk (in Parquet or CSV format)
+
+The following commands can be entered into spark-shell to register the data files that the 
+benchmark will query.
+
+```
+import com.nvidia.spark.rapids.tests.tpcds._
+TpcdsLikeSpark.setupAllParquet(spark, "/path/to/tpcds")
+```
+
+The benchmark can be executed with the following syntax to execute the query and collect the 
+results to the driver.
+
+```
+TpcdsLikeBench.collect(spark, "q5", iterations=3)
```

Comment:
did we make tpch and tpcxbb all common like this?  I used to run tpch like: Q2Like(spark).collect()?

Reply:
TPC-DS and TPCxBB are consistent so far. TPC-H has fallen behind a bit and I should probably update that too.

Reply:
I have now updated https://github.com/NVIDIA/spark-rapids/pull/823 to make TPC-H consistent.

Reply:
great thanks